yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -540,6 +561,29 @@ private[spark] class Client(
     amContainer
   }
 
+  def setupCredentials(): Unit = {
+    Option(args.principal) match {
```

Comment:
I agree.  Odersky himself as argued against using `Option(maybeNull)` if you are just going to immediately pattern match on it.  `if / else` would probably be simpler here.

I'm not recommending it in this case, but just fyi you can also match on multiple things at once:

```
(Option(args.principal), Option(args.keytab)) match {
  case (Some(principal), Some(keytabPath)) => ...
  case (Some(principal), None) => throw new SparkException( ...)
  case _ => credentials = UserGroupInformation.getCurrentUser.getCredentials
}
```